### PR TITLE
SplitIndex flyout change to page

### PR DESCRIPTION
**Pull request by xluo-aws**
### Description
[Refactor Split Index from flyout to new page; Move common functions to helper class]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by SuZhou-Joe** [comment]
public/pages/IndexDetail/containers/IndexDetail/IndexDetail.test.tsx

This test do not pass, please leave the snapshot as unchanged because the github unit test runner is in different timezone.

**Comment by xluo-aws** [comment]
> public/pages/IndexDetail/containers/IndexDetail/IndexDetail.test.tsx
> 
> This test do not pass, please leave the snapshot as unchanged because the github unit test runner is in different timezone.

Got it, will revert change to this file

**Comment by Hailong-am** [comment]
you could do `export TZ=UTC` first and then ran `yarn test:jest`, that will solve timezone issue.

File: public/pages/SplitIndex/container/SplitIndex/__snapshots__/SplitIndex.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      >
+        Split index
+      </h1>
+      <div
+        class="euiSpacer euiSpacer--l"
+      />
+    </div>
+  </div>,
+]
+`;
```
Comment (Hailong-am) [inline public/pages/SplitIndex/container/SplitIndex/__snapshots__/SplitIndex.test.tsx.snap:20]: why the snapshot only contains a title, should there have some content?

File: public/pages/SplitIndex/container/SplitIndex/SplitIndex.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  async componentDidMount() {
+    await this.isSourceIndexReady();
+    this.calculateShardsOption();
+    this.setState({
+      splitIndexFlyoutVisible: true,
+    });
+  }
+
+  isSourceIndexReady = async () => {
+    const source = queryString.parse(this.props.location.search) as { source?: string };
```
Comment (Hailong-am) [inline public/pages/SplitIndex/container/SplitIndex/SplitIndex.tsx:54]: what's the behavior when query string don't have source parameter? let's say user type the URL directly in browser.

**Comment by xluo-aws** [comment]
> you could do `export TZ=UTC` first and then ran `yarn test:jest`, that will solve timezone issue.

Good to know, Thanks

File: public/pages/SplitIndex/components/SplitIndexFlyout/SplitIndexFlyout.tsx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+interface SplitIndexComponentProps {
+  sourceIndex: string;
+  reasons: React.ReactChild[];
+  shardsSelectOptions: { label: string }[];
+  onSplitIndex: (targetIndex: string, settingsPayload: Required<IndexItem>["settings"]) => Promise<void>;
+  onCancel: () => void;
+  getAlias: AliasSelectProps["refreshOptions"];
+}
+
+export default class SplitIndexFlyout extends Component<SplitIndexComponentProps> {
```
[the remark this answers is not in the document]
Comment (xluo-aws) [reply public/pages/SplitIndex/components/SplitIndexFlyout/SplitIndexFlyout.tsx:28]: Yes, the name is not appropriate,  I use SplitIndex for container, but haven't thought a good name for component.

File: public/pages/SplitIndex/container/SplitIndex/SplitIndex.tsx (hunk 2 above)
Comment (xluo-aws) [reply public/pages/SplitIndex/container/SplitIndex/SplitIndex.tsx:54]: User will get a blank page and an error saying that not able to get index setting

File: public/pages/SplitIndex/container/SplitIndex/__snapshots__/SplitIndex.test.tsx.snap (hunk 1 above)
Comment (xluo-aws) [reply public/pages/SplitIndex/container/SplitIndex/__snapshots__/SplitIndex.test.tsx.snap:20]: Good catch, I didn't pay attention to snapshot file, as long as it didn't report an error, will take a look at it later.

**Review by SuZhou-Joe — APPROVED** [review]

**Review by Hailong-am — APPROVED** [review]

**Pull request merged**